packages/web/pages/search/players.tsx

```diff
@@ -0,0 +1,75 @@
+import { Text, VStack } from '@metafam/ds';
+import { PageContainer } from 'components/Container';
+import { PlayerList } from 'components/Player/PlayerList';
+import SearchFilters from 'components/SearchFilters';
+import { HeadComponent } from 'components/Seo';
+import { usePlayerFilter } from 'lib/hooks/player/players';
+import { useOnScreen } from 'lib/hooks/useOnScreen';
+import { useRouter } from 'next/router';
+import { MorePlayers } from 'pages/players';
+import React, { useEffect, useMemo, useRef } from 'react';
+import { GlobalFilters } from 'utils/GlobalSearch';
+
+const PlayerSearchPage: React.FC = () => {
+  const {
+    players,
+    fetching,
+    fetchingMore,
+    error,
+    queryVariables,
+    setQueryVariable,
+    total,
+    nextPage,
+    moreAvailable,
+  } = usePlayerFilter();
+  const { query } = useRouter();
+  const searchQuery = (Array.isArray(query.q) ? query.q[0] : query.q) || '';
```

Comment:
Your suggestion would give type error.
query.q could be a string or array

Reply:
> `query.q` could be a string or array

`Array.from` will return an array regardless of if `query.q`'s a string or array. If it's a string it'll return an array with that string as the single entry, or if it's an array it'll just return that array unmodified.

I'll go check the typing…

Reply:
> Your suggestion would give type error. `query.q` could be a string or array

@Udit-takkar, you're right, there is a type error, but not because `query.q` can be a string or an array, it's because it can be `undefined`.

This statement works without complaint: `const searchQuery = Array.from(query.q ?? '')[0]`.